dom.go

```diff
@@ -849,14 +849,12 @@ func (b *batchRenderer) render(startTime float64) {
 		}
 
 		// Perform render.
-		prevRender := c.Context().prevRender
-		nextRender, skip, pendingMounts := renderComponent(c, prevRender)
+		prevHTML := extractHTML(c.Context().prevRender)
+		nextHTML, skip, pendingMounts := renderComponent(c, c)
```

Comment:
This was the source of my bug - always should have been passing the component here instead of `prevRender`, since this is what we're actually doing.  The new code and tests exposed this bug, as `prevRender` can be a component (of the same type) now.